src/responder/nss/nsssrv.c

```diff
@@ -540,6 +541,20 @@ int main(int argc, const char *argv[])
     /* set up things like debug, signals, daemonization, etc... */
     debug_log_file = "sssd_nss";
 
+    if (is_socket_activated()) {
+        char *logpath;
+        int fd;
+
+        ret = asprintf(&logpath, "%s/%s.log", LOG_PATH, debug_log_file);
+        if (ret == -1) {
+            return ENOMEM;
+        }
+
+        fd = open(logpath, O_RDONLY);
+        fchown(fd, 0, 0);
+        close(fd);
+    }
+
```

Comment:
The whole point of using `fchown()` would be to avoid a time-of-check-time-of-use bug. If you close the file, then it could change again before it's opened for use.

This should be implemented as part of `open_debug_file_ex()` in debug.c